### DISCUSSION TITLE
Help Needed: 500 Server Error with Framework Read API

### DISCUSSION THREAD

**Discussion by rajnishdargan** [Sunbird-Knowlg/Community#216] [General Discussions]
Hello Team,

Our team recently set up Sunbird using the Sunbird Installer and installed version 7.0.0. We have the Knowlg (taxonomy, content, etc.) and inQuiry (assessment service) services deployed

I successfully created a channel using the /channel/v3/create endpoint with the following payload:
`{
  "request": {
    "channel": {
      "name": "Test Channel",
      "code": "test-channel"
    }
  }
}
`

API Documentation: [Channel Create API](https://documenter.getpostman.com/view/25463377/2s9YC4VDFm#b150b829-350f-4794-9341-0ca097528f8c)

Next, I attempted to create a framework using the /framework/v3/create endpoint with this payload:
`{
    "request": {
        "framework": {
            "name": "Test Framework",
            "code": "test_framework",
            "description": "Framework for testing",
            "type": "K-12",
            "channels": [
                {
                    "identifier": "test-channel"
                }
            ]
        }
    }
}
`

API Documentation: [Framework Create API](https://documenter.getpostman.com/view/25463377/2s9YC4VDL6#6c1af6c3-7b25-4293-9896-b143a25bee15)

The framework creation API returns a 200 success status. However, when trying to read the framework using the /framework/v3/read/{frameworkCode} endpoint, I receive a 500 Server Error. The logs show the following error:

`com.typesafe.config.ConfigException$Missing: No configuration setting found for key 'external.primaryKey'
...
...
Exception occurred - class :com.typesafe.config.ConfigException$Missing with message :No configuration setting found for key 'external.primaryKey'
`
Framework Read URL: {host}/framework/v3/read/test_framework

Can anyone help me resolve this issue? It appears to be a configuration problem, but I’m not sure how to address it.

cc  @coolbung @snehal0904 @AbhilashKD

**Comment by coolbung** [comment]
Could you help with this @HarishGangula @maheshkumargangula @pallakartheekreddy

**Comment by aimansharief** [comment]
@rajnishdargan Can you please share the link that you referred and installed Sunbird. Also can you please more detailed error log from the `framework/v3/read` API.

**Reply by AbhilashKD** [reply]
@aimansharief 
We have refereed below repo with branch release-6.0.0

Link : https://github.com/project-sunbird/sunbird-ed-installer.git

**Reply by aimansharief** [reply]
@AbhilashKD If you're using `release-6.0.0`, you should direct your requests to the `learning-service` for the framework APIs. Before testing the APIs, ensure that the learning-service, Cassandra, and Neo4j are all up and running.

**Reply by AbhilashKD** [reply]
@aimansharief 
Apologies for the confusion 
we are using release-7.0.0 of the sunbird-ed-installer 
As our Infra Hosted on AWS So did some minor changes to deploy the services on AWS Cloud under EKS Cluster

**Reply by aimansharief** [reply]
@AbhilashKD 
In `release-7.0.0` you should be pointing it to `taxonomy-service` itself. The Error which is mentioned above is because of the Incorrect Schema. Verify your schema upload step, you might have uploaded old schemas.

**Reply by AbhilashKD** [reply]
@aimansharief 
Can you tell me Where is the schema upload step documented?
I have not seen any documentations regarding changing schemas from old to new.

**Reply by aimansharief** [reply]
@AbhilashKD In the ed-installer they might have automated to upload the schemas or getting it from blob. Since you are trying to deploy on AWS, I'm just assuming it might have got missed.

For Framework this is the expected schema and config 
https://github.com/Sunbird-Knowlg/knowledge-platform/tree/release-5.7.0_RC11/schemas/framework/1.0

**Comment by rajnishdargan** [comment]
Hi @aimansharief,
Thank you for your support. Here is the complete error log: 
[sunbird-taxonomy-79dfcdbdb8-dmpp2-1724765046324397441.log](https://github.com/user-attachments/files/16762791/sunbird-taxonomy-79dfcdbdb8-dmpp2-1724765046324397441.log)
 
We will get back to you shortly with the details of the installation steps.

**Comment by pallakartheekreddy** [comment]
@rajnishdargan As we haven't received a response from you, we will proceed to close this thread. If you have any further questions or require assistance in the future, please don't hesitate to start a new thread. Thank you.
